cli/src/commands/file/annotate.rs

```diff
@@ -120,18 +120,22 @@ fn render_file_annotation(
     let mut formatter = ui.stdout_formatter();
     let mut last_id = None;
     let default_id = repo.store().root_commit_id();
-    for (line_number, (commit_id, content)) in annotation.lines().enumerate() {
+    for (line_number, (line_origin, content)) in annotation.line_origins().enumerate() {
         /* At least in cases where the repository was jj-initialized shallowly,
         then unshallow'd with git, some changes will not have a commit id
         because jj does not import the unshallow'd commits. So we default
         to the root commit id for now. */
-        let commit_id = commit_id.unwrap_or(default_id);
+        let (commit_id, original_line_number) = match line_origin {
+            Ok(origin) => (&origin.commit_id, origin.line_number),
+            Err(_) => (default_id, 0),
+        };
```

Comment:
nit: we could use `line_origin.unwrap_or(&default_line_origin)` or something.

Reply:
Yes, I meant that. It is a nasty hack to hide data corruption of unshallow-ed repository, and we'll need to remove `default_line_origin` at some point.

Reply:
Yes, sounds good to me.